requirement-generator/checkers/majors/cs.ts

```diff
@@ -0,0 +1,33 @@
+import { Course } from '../../types';
+import { courseMatchesCode, courseMatchesCodeOptions } from '../checkers-common';
+
+const csIntroProgramming = (course: Course): boolean => (
+  courseMatchesCodeOptions(course, ['CS 1110', 'CS 1112', 'CS 1114', 'CS 1115'])
+  || courseMatchesCodeOptions(course, ['CS 2110', 'CS 2112'])
```

Comment:
What's the plan to not allow for cases of cross-over requirements like CS 1110 and CS 1112 satisfying the entire csIntroProgramming requirement?

Reply:
It's not achievable with the current setup. The current setup simply exposes the problem more clearly. However, I do keep the group structure so that we do not completely lose the original json structure.

Reply:
That makes sense